liquibase-core/src/main/java/liquibase/util/StreamUtil.java

```diff
@@ -99,12 +99,35 @@ public static String getReaderContents(Reader reader) throws IOException {
     public static void copy(InputStream inputStream, OutputStream outputStream) throws IOException {
         byte[] bytes = new byte[1024];
         int r = inputStream.read(bytes);
-        while (r > 0) {
+        while (r > 0 ) {
             outputStream.write(bytes, 0, r);
             r = inputStream.read(bytes);
         }
     }
 
+    /**
+     * Writes 1Mb (in case of size wasn't specified) inputStream to outputStream
```

Comment:
This says the block is 1 MB, but it appears to be 1024 bytes.

Reply:
I thought here we have:

1. We get inputStream. It can be Gbs of data.
2. Set size. [In current case it's 1024 Kb](https://github.com/Datical/liquibase/pull/78/files#diff-6db4d1e1bd4a92a2c2c1e9ce97fb59a8R373) 
3. Iterate over inputStream 1024 times
and eatch time we take 1024 bytes = 1kb 

I'm not sure I got what you're saying here.  What I missed in list upper ?

Reply:
It's not setting a 1MB block size, it's reading 1024 1k blocks and then not copying any more. 

I'm not sure arbitrarily cutting off the output is the best way to solve the memory issue vs. figuring out where the memory is actually being used. We are just writing to an output stream which should be more of a file-based process and not really be a cause of memory usage. I'm sure it's buffering some things, so maybe just make sure we flush the buffer every 5mb or something and see if that helps the memory problem?

Reply:
I moved/rewrote this code a bit which I think resolves these issues